### DISCUSSION TITLE
Valve config

### DISCUSSION THREAD

**Discussion by lukasa1993** [HomeSpan/HomeSpan#872] [Q&A] [answered]
hey i need a way to configure how long i open valve it there any good way to have variable from homekit ?

**Comment by HomeSpan** [comment]
Lots of ways of doing this.  If you are using the Irrigation Service, you can have timers set and displayed in the Home App.  See the [Irrigation Reference Sketch](https://github.com/HomeSpan/HomeSpanReferenceSketches/blob/main/Irrigation/Irrigation.ino) for a comprehensive example.

For a more simple control of a Valve (open, wait some time, then close), use the same logic that applies to any Characteristic when you want an action to be followed by a subsequent action at a later time using `timeVal()`.   See Tutorial 14 for details:  https://github.com/HomeSpan/HomeSpan/tree/master/examples/14-EmulatedPushButtons

**Reply by lukasa1993** [reply]
yea i am using timeVal and for hardcoded values everything works, i just want to adjust how long it wats somehow

**Reply by HomeSpan** [reply]
To select open/close timing from the Home App you need to use the SetDuration and RemainingDuration Characteristics.  These are a little tricky to use but are detailed in the Irrigation Reference Sketch.  Unfortunately the selection screen in the Home App only provides a fixed set of durations - I've experimented a bunch but have not been able to find a way of customizing these values so that you can set your own set of selectable durations.

**Reply by lukasa1993** [reply]
any other easy waits outside of home app to do that ?

**Reply by HomeSpan** [reply, marked as answer]
One way would be from the Serial Monitor - you could create a simple user-defined command that accepts a timing duration number used by the app.  But this of course means your device needs to be connected to a computer.

Alternatively, you can add a web serving page to your device with an interface that allows you to enter new information that can be consumed by the sketch.   I included a few examples of web servers being used in conjunction with the HomeSpan, but you'll need to code the HTML and web forms directly to suit your specific need.
